lm_eval/models/optimum_lm.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+
+from lm_eval.api.registry import register_model
+from lm_eval.models.huggingface import HFLM
+
+
+@register_model("optimum-causal")
```

Comment:
`optimum` is a broader name that is also related to ONNX Runtime. I would rename it to `openvino-causal`. Please consider changing Readme and test accordingly. We can also consider implementing `openvino-seq2seq` in the future to support T5, Bart and similar models.

Reply:
@haileyschoelkopf What do you think about calling it "openvino" instead, to align with "hf" for example? When we add seq2seq models in a later PR, that should preferably work the same way as for "hf" models.

Reply:
That would be good with me!

Reply:
I've renamed it to `openvino` and added a more descriptive error message to the "causal" backend assert.